### PR TITLE
link editor

### PR DESCRIPTION
**Pull request by Pomax**
sets up the basics for fixing #1645, although it still needs to actually be hooked up to "do" things in conjunction with a real page.

### CODE REVIEW

File: app/src/main/java/mozilla/org/webmaker/WebmakerApplication.java [hunk 1]
```diff
@@ -32,7 +32,8 @@ public void onCreate() {
         Router.sharedRouter().map("/main/:tab", MainActivity.class);
         Router.sharedRouter().map("/map/:projectId", Map.class);
         Router.sharedRouter().map("/projects/:projectId", Project.class);
-        Router.sharedRouter().map("/projects/:projectId/elements/:elementId", Editor.class);
+        Router.sharedRouter().map("/edit/:projectId/:elementId/:type", Editor.class);
```
Comment (k88hudson) [inline app/src/main/java/mozilla/org/webmaker/WebmakerApplication.java]: while i agree these routes need some work, probably doing in another ticket -- would you mind holding off? (will reduce merge conflicts too)

File: www_src/pages/editor/link-editor.jsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+          <div className="form-group">
+            <label>Font</label>
+            <select className="select" valueLink={this.linkState('fontFamily')}>
+              <option value="Roboto">Roboto</option>
+              <option value="Bitter">Bitter</option>
+              <option value="Pacifico">Pacifico</option>
+            </select>
+          </div>
+          <div className="form-group">
+            <label>Color</label>
```
Comment (k88hudson) [inline www_src/pages/editor/link-editor.jsx]: Designs say "Background Color"

File: www_src/pages/editor/link-editor.jsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+var LinkEditor = React.createClass({
+  mixins: [
+    React.addons.LinkedStateMixin,
+    Binding
+  ],
+  getInitialState: function () {
+    return {
+      borderRadius: 5,
+      color: "#69A0FC",
+      textColor: "white",
```
Comment (k88hudson) [inline www_src/pages/editor/link-editor.jsx]: I think what we actually need here is to compute whether white or black has greater contrast from the background color -- there are a bunch of libraries that can handle this such as https://www.npmjs.com/package/color. It should be pretty simple to do. would you mind adding that in?

File: app/src/main/java/mozilla/org/webmaker/WebmakerApplication.java (hunk 1 above)
Comment (Pomax) [inline app/src/main/java/mozilla/org/webmaker/WebmakerApplication.java]: not really: the routing is necessary to have the correct element pull up the correct editor on Android, which cannot hook into window.location.has like editor.jsx does right now (That only works in the desktop browser)

File: www_src/pages/editor/link-editor.jsx (hunk 3 above)
Comment (Pomax) [inline www_src/pages/editor/link-editor.jsx]: agreed, will updated the code to make that happen automagically

File: www_src/pages/editor/link-editor.jsx (hunk 2 above)
Comment (Pomax) [inline www_src/pages/editor/link-editor.jsx]: whoops, good catch

File: www_src/lib/color.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+var Color = require('color');
+var minimumContrast = 2;
+
+module.exports = {
+  getContrastingColor: (function() {
```
Comment (k88hudson) [inline www_src/lib/color.js]: no need to reimplement .contrast https://www.npmjs.com/package/color#luminosity

**Comment by Pomax** [comment]
resetting to "needs review" because I'm fairly certain the rebase went fine, but I'd like another R? pass just to be sure.

**Pull request merged**